package/gluon-core/luasrc/lib/gluon/upgrade/110-network

```diff
@@ -4,6 +4,23 @@ local uci = require('simple-uci').cursor()
 
 local sysconfig = require 'gluon.sysconfig'
 local util = require 'gluon.util'
+local json = require 'jsonc'
+
+local function get_network_mac(name)
+	local board_data = json.load('/etc/board.json')
+	local network_data = (board_data or {}).network
+
+	if network_data == nil then
+		return nil
+	end
+
+	local ifdata = network_data[name] or {}
+	if ifdata.macaddr then
+		return ifdata.macaddr
+	end
+
+	return nil
```

Comment:
Just a nit: It would be nice if the final return would be the success case. The `if ifdata.macaddr` check is redundant anyways, as `return ifdata.macaddr` will return `nil` if the field is unset.

Reply:
Maybe it's just python speaking, but I like the explicit approach.
Anyway, the code looks fine. Has it been tested to work like intended?